src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -1386,6 +1389,10 @@ private void handlePutBlob(HttpServletRequest request,
                 }
             }
 
+            if (blobStoreType.equals("azureblob")) {
+                eTag = BaseEncoding.base16().lowerCase()
+                        .encode(BaseEncoding.base64().decode(contentMD5String));
+            }
```

Comment:
Thanks for the feedback. I could be wrong about this but the AWS Java SDK expects a base16 encoded ETag header. If server returns an ETag like #96 , Client app will fail. While MD5 header is returned to the client for PUT, unless client app does extra work to validate, the current Java SDK will fail with invalid value returned for ETag. I can definitely add checks for null MD5 client header.